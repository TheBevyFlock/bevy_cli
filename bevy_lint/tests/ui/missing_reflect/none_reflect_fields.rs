//! Tests the `missing_reflect` lint when `Component`, `Resource`, and `Event` have fields that do
//! not implement `Reflect`

#![feature(register_tool)]
#![register_tool(bevy)]
#![allow(dead_code)]
//~v NOTE: the lint level is defined here
#![deny(bevy::missing_reflect)]

use bevy::prelude::*;

struct NonReflect(u64);

//~v NOTE: `Component` implemented here
#[derive(Component)]
//~| HELP: `Reflect` can be automatically derived
//~v ERROR: defined a component without a `Reflect` implementation
struct MyComponent {
    reflect: u64,
    non_reflect: NonReflect,
}

//~v NOTE: `Resource` implemented here
#[derive(Resource)]
//~| HELP: `Reflect` can be automatically derived
//~v ERROR: defined a resource without a `Reflect` implementation
enum MyResource {
    Reflectable(String),
    NonReflectable(NonReflect),
}
//~| HELP: `Reflect` can be automatically derived
//~v ERROR: defined an event without a `Reflect` implementation
struct MyEvent(NonReflect);
//~v NOTE: `Event` implemented here
impl Event for MyEvent {
    type Traversal = ();
}
