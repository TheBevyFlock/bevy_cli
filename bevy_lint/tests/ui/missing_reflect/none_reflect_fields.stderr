error: defined an event without a `Reflect` implementation
  --> tests/ui/missing_reflect/none_reflect_fields.rs:33:1
   |
33 | struct MyEvent(NonReflect);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: `Event` implemented here
  --> tests/ui/missing_reflect/none_reflect_fields.rs:35:1
   |
35 | / impl Event for MyEvent {
36 | |     type Traversal = ();
37 | | }
   | |_^
note: the lint level is defined here
  --> tests/ui/missing_reflect/none_reflect_fields.rs:8:9
   |
8  | #![deny(bevy::missing_reflect)]
   |         ^^^^^^^^^^^^^^^^^^^^^
help: `Reflect` can be automatically derived
   |
33 + #[derive(Reflect)]
34 | struct MyEvent(NonReflect);
   |

error: defined a component without a `Reflect` implementation
  --> tests/ui/missing_reflect/none_reflect_fields.rs:18:1
   |
18 | / struct MyComponent {
19 | |     reflect: u64,
20 | |     non_reflect: NonReflect,
21 | | }
   | |_^
   |
note: `Component` implemented here
  --> tests/ui/missing_reflect/none_reflect_fields.rs:15:10
   |
15 | #[derive(Component)]
   |          ^^^^^^^^^
   = note: this error originates in the derive macro `Component` (in Nightly builds, run with -Z macro-backtrace for more info)
help: `Reflect` can be automatically derived
   |
18 + #[derive(Reflect)]
19 | struct MyComponent {
   |

error: defined a resource without a `Reflect` implementation
  --> tests/ui/missing_reflect/none_reflect_fields.rs:27:1
   |
27 | / enum MyResource {
28 | |     Reflectable(String),
29 | |     NonReflectable(NonReflect),
30 | | }
   | |_^
   |
note: `Resource` implemented here
  --> tests/ui/missing_reflect/none_reflect_fields.rs:24:10
   |
24 | #[derive(Resource)]
   |          ^^^^^^^^
   = note: this error originates in the derive macro `Resource` (in Nightly builds, run with -Z macro-backtrace for more info)
help: `Reflect` can be automatically derived
   |
27 + #[derive(Reflect)]
28 | enum MyResource {
   |

error: aborting due to 3 previous errors

